### PR TITLE
Create BareMetal cluster with user provided cluster_name

### PR DESCRIPTION
**Pull request by prsurve**
Signed-off-by: prsurve <prsurve@redhat.com>

### CODE REVIEW

File: ocs_ci/utility/aws.py [hunk 1]
```diff
@@ -1357,13 +1357,16 @@ def delete_cf_stack_including_dependencies(self, cfs_name):
...
+    def delete_hosted_zone(
+        self, cluster_name, delete_zone=True, delete_from_base_domain=False
+    ):
         """
         Deletes the hosted zone
 
         Args:
             cluster_name (str): Name of the cluster
             delete_zone (bool): Whether to delete complete zone
+            delete_from_base_domain (bool): Whether to delete record from base domain
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1369]: leave blank line after this line

File: ocs_ci/utility/aws.py [hunk 2]
```diff
@@ -1630,6 +1641,48 @@ def get_hosted_zone_id_for_domain(self, domain=None):
         ]
         return hosted_zone_ids[0]
 
+    def create_hosted_zone(self, cluster_name):
+        """
+        Create Hosted Zone
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1646]: leave blank line

File: ocs_ci/utility/aws.py [hunk 3]
```diff
@@ -1630,6 +1641,48 @@ def get_hosted_zone_id_for_domain(self, domain=None):
...
+        Args:
+            cluster_name (str): Name of cluster
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1648]: leave blank line

File: ocs_ci/utility/aws.py [hunk 4]
```diff
@@ -1630,6 +1641,48 @@ def get_hosted_zone_id_for_domain(self, domain=None):
...
+        Returns:
+            str: Hosted Zone id
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1650]: leave blank line

File: ocs_ci/utility/aws.py [hunk 5]
```diff
@@ -1630,6 +1641,48 @@ def get_hosted_zone_id_for_domain(self, domain=None):
...
+        )
+
+        full_hosted_zone_id = response["HostedZone"]["Id"]
+        hosted_zone_id = full_hosted_zone_id.strip("/hostedzone/")
+        logger.info(
+            f"Hosted zone Created with id {hosted_zone_id} and name is {response['HostedZone']['Name']}"
+        )
+        return hosted_zone_id
+
+    def get_hosted_zone_details(self, zone_id):
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1666]: doc string format (same as def create_hosted_zone )

File: ocs_ci/utility/aws.py [hunk 6]
```diff
@@ -1630,6 +1641,48 @@ def get_hosted_zone_id_for_domain(self, domain=None):
...
+        """
+        Get Hosted zone Details
+        Args:
+            zone_id (str): Zone Id of cluster_name
+        Returns:
+            dict: Response
+        """
+        return self.route53_client.get_hosted_zone(Id=zone_id)
+
+    def get_ns_for_hosted_zone(self, zone_id):
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1676]: doc string format (same as def create_hosted_zone )

File: ocs_ci/utility/aws.py (hunk 1 above)
Comment (prsurve) [reply ocs_ci/utility/aws.py:1369]: Done

File: ocs_ci/utility/aws.py (hunk 2 above)
Comment (prsurve) [reply ocs_ci/utility/aws.py:1647]: Done

File: ocs_ci/utility/aws.py (hunk 3 above)
Comment (prsurve) [reply ocs_ci/utility/aws.py:1650]: Done

File: ocs_ci/utility/aws.py (hunk 4 above)
Comment (prsurve) [reply ocs_ci/utility/aws.py:1653]: Done

File: ocs_ci/utility/aws.py (hunk 5 above)
Comment (prsurve) [reply ocs_ci/utility/aws.py:1670]: Done

File: ocs_ci/utility/aws.py (hunk 6 above)
Comment (prsurve) [reply ocs_ci/utility/aws.py:1683]: Done

**Review by vavuthu — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Pull request merged**